Play Services crash after installation

Play services crash after installation, the notification shows that it's running but for an example Signal doesn't see it at all and after installation it switches to being always on in the background.
I've reinstalled os multiple times via web installer, every time I get the same result.

If there's anything I could do more, let me know.

The error code:

```
type: crash
osVersion: google/sunfish/sunfish:13/TQ3A.230805.001.S1/2024010700:user/release-keys
package: com.google.android.gms:234914038
process: com.google.android.gms
processUptime: 4780 + 1178 ms
installer: app.grapheneos.apps
GmsCompatConfig version: 93

java.lang.SecurityException: Attempt to change component state; pid=6534, uid=10148, package=com.google.android.gms.supervision
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3020)
	at android.os.Parcel.createException(Parcel.java:3004)
	at android.os.Parcel.readException(Parcel.java:2980)
	at android.os.Parcel.readException(Parcel.java:2922)
	at android.content.pm.IPackageManager$Stub$Proxy.setApplicationEnabledSetting(IPackageManager.java:6080)
	at android.app.ApplicationPackageManager.setApplicationEnabledSetting(ApplicationPackageManager.java:3084)
	at com.android.internal.gmscompat.sysservice.GmcPackageManager.setApplicationEnabledSetting(GmcPackageManager.java:138)
	at bgqv.a(:com.google.android.gms@234914038@23.49.14 (190400-590296185):12)
	at com.google.android.gms.kids.settings.KidsSettingsModuleInitIntentOperation.onHandleIntent(:com.google.android.gms@234914038@23.49.14 (190400-590296185):163)
	at com.google.android.chimera.IntentOperation.onHandleIntent(:com.google.android.gms@234914038@23.49.14 (190400-590296185):2)
	at agxh.onHandleIntent(:com.google.android.gms@234914038@23.49.14 (190400-590296185):10)
	at nxx.run(:com.google.android.gms@234914038@23.49.14 (190400-590296185):70)
	at nxw.run(:com.google.android.gms@234914038@23.49.14 (190400-590296185):152)
	at dyxb.run(:com.google.android.gms@234914038@23.49.14 (190400-590296185):13)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.PackageManagerService.setEnabledSettings(PackageManagerService.java:3743)
	at com.android.server.pm.PackageManagerService.-$$Nest$msetEnabledSettings(Unknown Source:0)
	at com.android.server.pm.PackageManagerService$IPackageManagerImpl.setApplicationEnabledSetting(PackageManagerService.java:5493)
	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:3025)
	at com.android.server.pm.PackageManagerService$IPackageManagerImpl.onTransact(PackageManagerService.java:5994)


```

--- a/core/java/android/content/ContentResolver.java
+++ b/core/java/android/content/ContentResolver.java
@@ -1247,6 +1247,10 @@ public abstract class ContentResolver implements ContentInterface {
                         queryArgs, remoteCancellationSignal);
             }
             if (qCursor == null) {
+                if (GmsCompat.isEnabled()) {
+                    return GmsHooks.maybeModifyQueryResult(uri, projection, queryArgs, null);
+                }
+
                 return null;
             }
 

--- a/core/java/com/android/internal/gmscompat/GmsHooks.java
+++ b/core/java/com/android/internal/gmscompat/GmsHooks.java
@@ -337,8 +337,8 @@ public final class GmsHooks {
     }
 
     // ContentResolver#query(Uri, String[], Bundle, CancellationSignal)
-    public static Cursor maybeModifyQueryResult(Uri uri, @Nullable String[] projection, @Nullable Bundle queryArgs,
-                                                Cursor origCursor) {
+    public static Cursor maybeModifyQueryResult(Uri uri,
+            @Nullable String[] projection, @Nullable Bundle queryArgs, @Nullable Cursor origCursor) {
         String uriString = uri.toString();
 
         Consumer<ArrayMap<String, String>> mutator = null;
@@ -417,18 +417,21 @@ public final class GmsHooks {
         }
 
         if (mutator != null) {
-            return modifyKvCursor(origCursor, mutator);
+            return modifyKvCursor(origCursor, projection, mutator);
         }
 
         return null;
     }
 
-    private static Cursor modifyKvCursor(Cursor origCursor, Consumer<ArrayMap<String, String>> mutator) {
+    private static Cursor modifyKvCursor(@Nullable Cursor origCursor, @Nullable String[] projection,
+                                         Consumer<ArrayMap<String, String>> mutator) {
         final int keyIndex = 0;
         final int valueIndex = 1;
         final int projectionLength = 2;
 
-        String[] projection = origCursor.getColumnNames();
+        if (origCursor != null) {
+            projection = origCursor.getColumnNames();
+        }
 
         boolean expectedProjection = projection != null && projection.length == projectionLength
                 && "key".equals(projection[keyIndex]) && "value".equals(projection[valueIndex]);
@@ -438,14 +441,18 @@ public final class GmsHooks {
             return null;
         }
 
-        ArrayMap<String, String> map = new ArrayMap<>(origCursor.getColumnCount() + 10);
+        final ArrayMap<String, String> map;
+        if (origCursor == null) {
+            map = new ArrayMap<>();
+        } else {
+            map = new ArrayMap<>(origCursor.getColumnCount() + 10);
+            try (Cursor orig = origCursor) {
+                while (orig.moveToNext()) {
+                    String key = orig.getString(keyIndex);
+                    String value = orig.getString(valueIndex);
 
-        try (Cursor orig = origCursor) {
-            while (orig.moveToNext()) {
-                String key = orig.getString(keyIndex);
-                String value = orig.getString(valueIndex);
-
-                map.put(key, value);
+                    map.put(key, value);
+                }
             }
         }
 

--- a/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
+++ b/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
@@ -24,6 +24,7 @@ import android.app.Application;
 import android.app.ApplicationPackageManager;
 import android.app.compat.gms.GmsCompat;
 import android.content.ComponentName;
+import android.content.ContentResolver;
 import android.content.Context;
 import android.content.Intent;
 import android.content.pm.ApplicationInfo;
@@ -35,8 +36,10 @@ import android.content.pm.PackageInfo;
 import android.content.pm.SharedLibraryInfo;
 import android.content.pm.VersionedPackage;
 import android.ext.PackageId;
+import android.os.Build;
 import android.os.Process;
 import android.os.UserHandle;
+import android.provider.Settings;
 import android.util.ArrayMap;
 import android.util.ArraySet;
 import android.util.Log;
@@ -364,7 +367,7 @@ public class GmcPackageManager extends ApplicationPackageManager {
             throw e;
         }
 
-        if (PackageId.ANDROID_AUTO_NAME.equals(packageName)) {
+        if (PackageId.ANDROID_AUTO_NAME.equals(packageName) && isUpdateOwnershipOverrideEnabled()) {
             // Android Auto needs to be exempted from updates via Play Store to prevent breaking the
             // compatibility layer support for Android Auto.
             //
@@ -382,6 +385,15 @@ public class GmcPackageManager extends ApplicationPackageManager {
         }
 
         return res;
+    }
+
+    private static boolean isUpdateOwnershipOverrideEnabled() {
+        if (!Build.isDebuggable()) {
+            return true;
+        }
+
+        ContentResolver cr = GmsCompat.appContext().getContentResolver();
+        return Settings.Global.getInt(cr, "gmscompat_allow_unknown_updates", 0) == 0;
     }
 
     private PackageInfo makePseudoDisabledPackageInfoOrThrow(String pkgName, PackageInfoFlags flags) throws NameNotFoundException {

--- a/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
+++ b/core/java/com/android/internal/gmscompat/sysservice/GmcPackageManager.java
@@ -139,7 +139,11 @@ public class GmcPackageManager extends ApplicationPackageManager {
             return;
         }
 
-        super.setApplicationEnabledSetting(packageName, newState, flags);
+        try {
+            super.setApplicationEnabledSetting(packageName, newState, flags);
+        } catch (SecurityException e) {
+            Log.d(TAG, "", e);
+        }
     }
 
     @Override

